src/beanmachine/paic2/include/WorldSpec.h

```diff
@@ -0,0 +1,35 @@
+//
+// Created by Steffi Stumpos on 7/19/22.
+//
+
+#ifndef PAIC2_WORLDSPEC_H
+#define PAIC2_WORLDSPEC_H
+#include <string>
+
+namespace paic2 {
+
+    // The compiler knows that the world has certain members and the layout of those members, it just doesn't know what those
+    // members are called. This is the configuration for the world fields.
+    class WorldSpec {
```

Comment:
this is a bit redundant. You exposed the built in world type in PaicAST. A configuration should not also still be necessary. Right now this is still necessary because a string is being used to map the python MetaWorld methods to the MLIR ops. You have two options: (1) expose the world interface in PaicAST. Python will map from strings to metadata types (2) extend the WorldSpec to enable configuring the rest of the world interface